app/src/main/java/org/schabi/newpipe/player/ui/VideoPlayerUi.java

```diff
@@ -0,0 +1,1578 @@
+    }
+
+    private void updateStreamRelatedViews() {
+        //noinspection SimplifyOptionalCallChains
+        if (!player.getCurrentStreamInfo().isPresent()) {
+            return;
+        }
+        final StreamInfo info = player.getCurrentStreamInfo().get();
+
+        binding.qualityTextView.setVisibility(View.GONE);
+        binding.playbackSpeed.setVisibility(View.GONE);
+
+        binding.playbackEndTime.setVisibility(View.GONE);
+        binding.playbackLiveSync.setVisibility(View.GONE);
+
+        switch (info.getStreamType()) {
+            case AUDIO_STREAM:
+                binding.surfaceView.setVisibility(View.GONE);
+                binding.endScreen.setVisibility(View.VISIBLE);
+                binding.playbackEndTime.setVisibility(View.VISIBLE);
+                break;
+
+            case AUDIO_LIVE_STREAM:
+                binding.surfaceView.setVisibility(View.GONE);
+                binding.endScreen.setVisibility(View.VISIBLE);
+                binding.playbackLiveSync.setVisibility(View.VISIBLE);
+                break;
+
+            case LIVE_STREAM:
+                binding.surfaceView.setVisibility(View.VISIBLE);
+                binding.endScreen.setVisibility(View.GONE);
+                binding.playbackLiveSync.setVisibility(View.VISIBLE);
+                break;
+
+            case VIDEO_STREAM:
+                //noinspection SimplifyOptionalCallChains
+                if (player.getCurrentMetadata() != null
+                        && !player.getCurrentMetadata().getMaybeQuality().isPresent()
+                        || (info.getVideoStreams().isEmpty()
+                        && info.getVideoOnlyStreams().isEmpty())) {
+                    break;
+                }
+
+                buildQualityMenu();
+
+                binding.qualityTextView.setVisibility(View.VISIBLE);
+                binding.surfaceView.setVisibility(View.VISIBLE);
+                // fallthrough
+            default:
+                binding.endScreen.setVisibility(View.GONE);
+                binding.playbackEndTime.setVisibility(View.VISIBLE);
+                break;
+        }
+
+        buildPlaybackSpeedMenu();
+        binding.playbackSpeed.setVisibility(View.VISIBLE);
+    }
+    //endregion
+
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Popup menus ("popup" means that they pop up, not that they belong to the popup player)
+    //////////////////////////////////////////////////////////////////////////*/
+    //region Popup menus ("popup" means that they pop up, not that they belong to the popup player)
+
+    private void buildQualityMenu() {
+        if (qualityPopupMenu == null) {
+            return;
+        }
+        qualityPopupMenu.getMenu().removeGroup(POPUP_MENU_ID_QUALITY);
+
+        @Nullable final List<VideoStream> availableStreams
+                = Optional.ofNullable(player.getCurrentMetadata())
```

Comment:
I think the ``=`` should be one line higher

Reply:
Then please open a checkstyle PR adding such a rule, this repo is full of things like this (many done by me). If there is nothing enforcing it, it is a waste of time to change it, just to forget next time ;-)

Reply:
@Stypox 
Created #8601